theano/gof/cmodule.py

```diff
@@ -2022,24 +2023,25 @@ def join_options(init_part):
                     _logger.info("g++ -march=native equivalent flags: %s",
                                  GCC_compiler.march_flags)
 
+            # Find working march flag:
+            default_detected_flag = GCC_compiler.march_flags[0][7:]
+            march_flags_to_try = [default_detected_flag, 'corei7-avx', 'corei7', 'core2']
+            for march_flag in march_flags_to_try:
+                test_flags = GCC_compiler.march_flags[:]
+                test_flags[0] = '-march=' + march_flag
+                result = try_march_flag(test_flags)
+                if result[0] and result[1]:
+                    GCC_compiler.march_flags[0] = march_flag
+                    GCC_compiler.march_flags[-1] = '-mtune=' + march_flag
+                    break
+            if not result[0] or not result[1]:
+                sys.exit('Could not find a working march')
```

Comment:
If we don't find any, we should not use one. Theano need to continue to
work on other type of CPU like arm CPU.
Le 8 avr. 2016 11:59, "ballasn" <notifications@github.com> a écrit :

> In theano/gof/cmodule.py
> <https://github.com/Theano/Theano/pull/4232#discussion_r59047573>:
>
> > @@ -2022,24 +2023,25 @@ def join_options(init_part):
> >                      _logger.info("g++ -march=native equivalent flags: %s",
> >                                   GCC_compiler.march_flags)
> >
> > +            # Find working march flag:
> > +            default_detected_flag = GCC_compiler.march_flags[0][7:]
> > +            march_flags_to_try = [default_detected_flag, 'corei7-avx', 'corei7', 'core2']
> > +            for march_flag in march_flags_to_try:
> > +                test_flags = GCC_compiler.march_flags[:]
> > +                test_flags[0] = '-march=' + march_flag
> > +                result = try_march_flag(test_flags)
> > +                if result[0] and result[1]:
> > +                    GCC_compiler.march_flags[0] = march_flag
> > +                    GCC_compiler.march_flags[-1] = '-mtune=' + march_flag
> > +                    break
> > +            if not result[0] or not result[1]:
> > +                sys.exit('Could not find a working march')
>
> This is a little bit drastic, we should raise a ValueError instead.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4232/files/9938f68d9fc9b44a14c4a209d9abeb983cd80c7c..bc6f5f0b26012365bb5c5c46571c9facad46c20e#r59047573>
>